src/error.rs

```diff
@@ -24,224 +27,153 @@
 ///
 /// Uses the error-chain crate to allow errors to allow more informative backtraces through error chaining.
 /// https://docs.rs/error-chain/0.12.2/error_chain/
+use thiserror::Error;
+use uuid::Uuid;
+
 pub mod errors {
-    use crate::sacn_parse_pack_error::sacn_parse_pack_error;
+    use super::*;
+
+    pub type Result<T> = std::result::Result<T, SacnError>;
 
-    /// UUID library used to handle the UUID's used in the CID fields, used here so that error can include the cid in messages.
-    use uuid::Uuid;
+    pub trait IntoSacnError {
+        fn into_sacn_error(self, context: String) -> SacnError;
+    }
 
-    error_chain! {
-        foreign_links {
-            Io(::std::io::Error);       // Allow IO errors to be used with the error-chain system.
-            Str(::std::str::Utf8Error); // Allow standard string library errors to be used with the error-chain system.
-            Uuid(uuid::Error);     // Allow UUID library to be used with error-chain system.
+    impl IntoSacnError for std::str::Utf8Error {
+        fn into_sacn_error(self, context: String) -> SacnError {
+            SacnError::Utf8Error(format!("{}: {}", context, self))
         }
+    }
 
-        links {
-            // All parse/pack errors live within the same chain ('family') of errors as described in sacn_parse_packet_error.
-            SacnParsePackError(sacn_parse_pack_error::Error, sacn_parse_pack_error::ErrorKind);
+    impl IntoSacnError for uuid::Error {
+        fn into_sacn_error(self, context: String) -> SacnError {
+            SacnError::UuidError(format!("{}: {}", context, self))
         }
+    }
```

Comment:
It would make sense to remove the UuidError and replace it with something specific that actually expresses what the uuid was being used for